src/supervisor-api.ts

```diff
@@ -114,10 +114,33 @@ export class SupervisorAPI {
 
 		this.api.use(authenticate(this.config));
 
-		this.api.post('/v1/blink', (_req, res) => {
+		this.api.post('/v1/blink', (req, res) => {
+			const blinks = checkInt(req.body.blinks) || undefined;
+			const onDuration = checkInt(req.body.onDuration) || undefined;
+			const offDuration = checkInt(req.body.offDuration) || undefined;
+			const pause = checkInt(req.body.pause) || undefined;
+			const timeout = checkInt(req.body.timeout) || undefined;
 			this.eventTracker.track('Device blink');
-			blink.pattern.start();
-			setTimeout(blink.pattern.stop, 15000);
+			blink.pattern.start({
+				blinks,
+				onDuration,
+				offDuration,
+				pause,
+			});
+			if (
+				(blinks === undefined &&
+					onDuration === undefined &&
+					offDuration === undefined &&
+					pause === undefined) ||
+				timeout !== undefined
+			) {
+				setTimeout(blink.pattern.stop, timeout || 15000);
+			}
+			return res.sendStatus(200);
+		});
+
+		this.api.delete('/v1/blink', (_req, res) => {
+			blink.pattern.stop();
```

Comment:
Does this happily work if a pattern is not currently ongoing?

Reply:
Yes: https://github.com/balena-io/blinking/blob/master/index.coffee#L41